### PR TITLE
ls: Fix display of inodes and add allocation size feature ("-s", "--size")

### PR DESCRIPTION
**Pull request by kimono-koans**
- Add block size ("-s", "--size") feature
- Fix how both block size and inode are displayed.  Instead of printing through either display_long_item or display_file_name, I've moved into a display_more_info function, which could be useful to other "ls" features that print the same whether in Format::Long or not.
- Tests included to check proper alignment and block size values.

### CODE REVIEW

File: src/uu/ls/src/ls.rs [hunk 1]
```diff
@@ -91,6 +91,7 @@ pub mod options {
     }
 
     pub mod size {
+        pub static ALLOCATION_SIZE: &str = "s";
```
Comment (sylvestre) [inline src/uu/ls/src/ls.rs:94]:
```suggestion
        pub static ALLOCATION_SIZE: &str = "size";
```

File: src/uu/ls/src/ls.rs [hunk 2]
```diff
@@ -1175,6 +1181,12 @@ only ignore '.' and '..'.",
                 .value_name("COLS")
                 .takes_value(true),
         )
+        .arg(
+            Arg::new(options::size::ALLOCATION_SIZE)
+                .short('s')
+                .long("size")
```
Comment (sylvestre) [inline src/uu/ls/src/ls.rs:1187]:
```suggestion
                .long(options::size::ALLOCATION_SIZE)
```

**Comment by sylvestre** [comment]
Looks good, just a nit

File: src/uu/ls/src/ls.rs (hunk 1 above)
Comment (kimono-koans) [reply src/uu/ls/src/ls.rs]: Makes sense.  Thanks for the review!

File: src/uu/ls/src/ls.rs (hunk 2 above)
Comment (kimono-koans) [reply src/uu/ls/src/ls.rs]: Makes sense.  Thanks for the review!

File: src/uu/ls/src/ls.rs [hunk 3]
```diff
@@ -340,6 +349,33 @@ struct PaddingCollection {
 }
 
 impl Config {
+    // From sort.rs
```
Comment (tertsdiepraam) [inline src/uu/ls/src/ls.rs:352]: If this is used in both `sort` and `ls`, let's move it to uucore!

File: src/uu/ls/src/ls.rs [hunk 4]
```diff
@@ -474,6 +510,30 @@ impl Config {
...
+                    .unwrap()
+                    .eq(&OsString::from("1"))
+            {
+                POSIXLY_CORRECT_BLOCK_SIZE
+            } else {
+                DEFAULT_BLOCK_SIZE
+            }
+        } else {
+            DEFAULT_BLOCK_SIZE
+        };
```
Comment (tertsdiepraam) [inline src/uu/ls/src/ls.rs:535]:
```suggestion
       let block_size = if let Ok(size) = std::env::var("BLOCK_SIZE") {
            if let Some(parsed_block_size) = Config::parse_byte_count(&size) {
                parsed_block_size
            } else {
                DEFAULT_BLOCK_SIZE // this branch should probably throw an error
            }
        } else if let Ok(c) = std::env::var("POSIXLY_CORRECT") {
            if c == "true" || c == "1" {
                POSIXLY_CORRECT_BLOCK_SIZE
            } else {
                DEFAULT_BLOCK_SIZE
            }
        } else {
            DEFAULT_BLOCK_SIZE
        };

```

**Review by tertsdiepraam — COMMENTED** [review]
Nice! Just 2 small suggestions

File: src/uu/ls/src/ls.rs (hunk 3 above)
Comment (kimono-koans) [reply src/uu/ls/src/ls.rs]: We now do this pretty differently, but noted.

File: src/uu/ls/src/ls.rs (hunk 4 above)
Comment (kimono-koans) [reply src/uu/ls/src/ls.rs]: I've rearranged this a bit, but the parse_byte_count function is where errors are printed.  Notably, we print a parse error and continue, while GNU does not.

**Comment by kimono-koans** [comment]
> Nice! Just 2 small suggestions

Appreciate your suggestions.  I'm going to write a few tests, and would appreciate your help with those, as well.

File: src/uu/ls/src/ls.rs (hunk 4 above)
Comment (tertsdiepraam) [reply src/uu/ls/src/ls.rs]: Sorry, I don't understand. Are you just talking about the comment in the suggestion or does the suggestion change the behaviour?

File: src/uu/ls/src/ls.rs (hunk 3 above)
Comment (tertsdiepraam) [reply src/uu/ls/src/ls.rs]: Ah okay. Just for my understanding then: is it fundamentally different functionality or should sort also behave this way?

File: src/uu/ls/src/ls.rs (hunk 4 above)
Comment (kimono-koans) [reply src/uu/ls/src/ls.rs]: I'm sorry.  I'm saying a few things things: 1) this is now an Option<usize>, and the code has changed to match the behavior of an Option, and 2) an error is *already printed in this case/branch,* it's just printed in the parse function where the error occurs.

**Pull request closed without merging**